src/pack.rs

```diff
@@ -119,14 +148,51 @@ pub async fn pack(options: PackOptions) -> Result<()> {
 
     let mut conda_packages_from_lockfile: Vec<CondaBinaryData> = Vec::new();
     let mut pypi_packages_from_lockfile: Vec<PypiPackageData> = Vec::new();
+    let mut loaded_injected_packages: Vec<PathBuf> = Vec::new();
 
     for package in packages {
         match package {
             LockedPackageRef::Conda(CondaPackageData::Binary(binary_data)) => {
                 conda_packages_from_lockfile.push(binary_data.clone())
             }
-            LockedPackageRef::Conda(CondaPackageData::Source(_)) => {
-                anyhow::bail!("Conda source packages are not yet supported by pixi-pack")
+            LockedPackageRef::Conda(CondaPackageData::Source(source_data)) => {
+                let base_dir = options.manifest_path.parent().unwrap_or(Path::new("."));
+                let pkg_dir = base_dir.join(match &source_data.location {
+                    UrlOrPath::Path(p) => p.to_path().to_string(),
+                    UrlOrPath::Url(u) => {
+                        anyhow::bail!("Unexpected URL for local package source: {}", u)
+                    }
+                });
+                let manifest_path = pkg_dir.join("pixi.toml");
+                let pkg_name = source_data.package_record.name.as_normalized();
+                eprintln!("🔨 Building package from the source: {}", pkg_name);
+                let bar = ProgressReporter::new(1);
+                build_local_package(
+                    &manifest_path,
+                    output_folder.path(),
+                    options.platform.as_str(),
+                )
+                .await?;
```

Comment:
Maybe out of scope for this PR but I'm wondering whether it makes sense to parallelize this.
@hofer-julian do you know whether there might be concurrency issues when invoking pixi-build for multiple packages at once?